src/components/chart/state-history-chart-line.ts

```diff
@@ -328,23 +328,94 @@ class StateHistoryChartLine extends LitElement {
           }
         });
       } else if (domain === "humidifier") {
+        const hasAction = states.states.some(
+          (entityState) => entityState.attributes?.action
+        );
+        const hasCurrent = states.states.some(
+          (entityState) => entityState.attributes?.current_humidity
+        );
+
+        const hasHumidifying =
+          hasAction &&
+          states.states.some(
+            (entityState: LineChartState) =>
+              entityState.attributes?.action === "humidifying"
+          );
+        const hasDrying =
+          hasAction &&
+          states.states.some(
+            (entityState: LineChartState) =>
+              entityState.attributes?.action === "drying"
+          );
+
         addDataSet(
           `${this.hass.localize("ui.card.humidifier.target_humidity_entity", {
             name: name,
           })}`
         );
-        addDataSet(
-          `${this.hass.localize("ui.card.humidifier.on_entity", {
-            name: name,
-          })}`,
-          true
-        );
+
+        if (hasCurrent) {
+          addDataSet(
+            `${this.hass.localize(
+              "ui.card.humidifier.current_humidity_entity",
+              {
+                name: name,
+              }
+            )}`
+          );
+        }
+
+        // If action attribute is available, we used it to shade the area below the humidity.
+        // If action attribute is not available, we shade the area when the device is on
+        if (hasHumidifying) {
+          addDataSet(
+            `${this.hass.localize("ui.card.humidifier.humidifying", {
+              name: name,
+            })}`,
+            true,
+            computedStyles.getPropertyValue("--state-humidifier-humidify-color")
+          );
+        } else if (hasDrying) {
+          addDataSet(
+            `${this.hass.localize("ui.card.humidifier.drying", {
+              name: name,
+            })}`,
+            true,
+            computedStyles.getPropertyValue("--state-humidifier-dry-color")
+          );
+        } else {
+          addDataSet(
+            `${this.hass.localize("ui.card.humidifier.on_entity", {
+              name: name,
+            })}`,
+            true
+          );
```

Comment:
We must have the same logic as climate, if there is no action, we don't know if it's active or now so we must not color the area.

Reply:
This is for backwards compatibility. For humidifiers that does not expose `action` attribute, we shade when it is on and don't shade when it is off. This is the current behavior, and the description (legend) will indicate what this metric is about.

Reply:
Don't get me wrong, I am OK removing it, just wanted to double check if you are sure :)